test/e2e/framework/kcp.go

```diff
@@ -465,6 +465,22 @@ func newPersistentKCPServer(name, kubeconfigPath string) (RunningServer, error)
 	}, nil
 }
 
+// NewLogicalServer creates a workspace in the provided server and org
+// and creates a server fixture for the logical cluster that results.
+func NewLogicalServer(t *testing.T, server RunningServer, org string) RunningServer {
```

Comment:
I don't understand what a logical server is, from these docs.

Reply:
Is this a workload cluster?

Reply:
`RunningServer` has previously represented a KCP server process. I'm attempting to provide an implementation backed by an unschedulable workspace so that the syncer can target it as if it were a pcluster. I thought this would be obvious from the code it replaces in this PR, but I guess not. What would you prefer it be called?

Reply:
NewFakeWorkloadCluster or NewFakeComputeCluster or so.

Reply:
It's not a cluster, it's a server (as per the return type), so I'll try NewFakeWorkloadServer.

Reply:
Done.